### PR TITLE
Upmerge from 0.15 to master.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

File: src/python/grpcio/grpc/__init__.py [hunk 1]
```diff
@@ -1218,10 +1218,10 @@ def server(thread_pool, handlers=None):
     thread_pool: A futures.ThreadPoolExecutor to be used by the returned Server
       to service RPCs.
     handlers: An optional sequence of GenericRpcHandlers to be used to service
-      RPCs after the returned Server is started. These handlers need not be the
-      only handlers the returned Server will use to service RPCs; other
-      handlers may later be added to the returned Server by calling its
-      add_generic_rpc_handlers method any time before it is started.
+      RPCs after the returned Server is started. These andlers need not be the
```
Comment (kpayson64) [inline src/python/grpcio/grpc/__init__.py]: andlers->handlers

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -1093,7 +1093,23 @@ def _shut_down_legacy_server(legacy_server_port):
...
+    pass
+  else:
+    urllib2.urlopen(
+        'http://localhost:%d/quitquitquit' % legacy_server_port).read()
+
+
 def _start_port_server(port_server_port):
+  # Temporary patch to switch the port_server port
+  # see https://github.com/grpc/grpc/issues/7145
+  _shut_down_legacy_server(32767)
```
Comment (kpayson64) [inline tools/run_tests/run_tests.py]: Remove this, this addition, it was a one time patch that has been removed.

Comment (nicolasnoble) [inline tools/run_tests/run_tests.py]: Done.

File: src/python/grpcio/grpc/__init__.py (hunk 1 above)
Comment (nicolasnoble) [inline src/python/grpcio/grpc/__init__.py]: Done.

**Comment by kpayson64** [comment]
lgtm

**Comment by jcanizales** [comment]
This broke gRPC.podspec too, as correctly captured by the Travis failure. Please revert the file to its state before this merge.

**Comment by jcanizales** [comment]
I've disabled Travis for grpc/grpc until this is fixed, as it's broken in master for every pull request. Please fix ASAP or further breakages will go unnoticed.

**Pull request merged**